Sources/SwiftCurrent_SwiftUI/Views/WorkflowView.swift

```diff
@@ -0,0 +1,215 @@
+        }
+    }
+
+    private init(isPresented: Binding<Bool>,
+                 workflow: AnyWorkflow?,
+                 onFinish: [(AnyWorkflow.PassedArgs) -> Void],
+                 onAbandon: [() -> Void],
+                 passedArgs: AnyWorkflow.PassedArgs) {
+        _isPresented = isPresented
+        self.workflow = workflow
+        self.onFinish = onFinish
+        self.onAbandon = onAbandon
+        self.passedArgs = passedArgs
+    }
+
+    public var body: some View {
+        if isPresented {
+            VStack {
+                model.body
+            }
+            .onAppear {
+                guard !didLoad else { return }
+                didLoad = true
+                model.isPresented = $isPresented
+                model.onAbandon = onAbandon
+                workflow?.launch(withOrchestrationResponder: model,
+                                 passedArgs: passedArgs,
+                                 launchStyle: .new) { passedArgs in
+                    onFinish.forEach { $0(passedArgs) }
+                }
+            }
+            .onDisappear {
+                model.body = AnyView(EmptyView())
+            }
+            .onReceive(inspection.notice) { inspection.visit(self, $0) } // Needed for ViewInspector
+        }
+    }
+
+    /// Adds an action to perform when this `Workflow` has finished.
+    public func onFinish(closure: @escaping (AnyWorkflow.PassedArgs) -> Void) -> Self {
+        var onFinish = self.onFinish
+        onFinish.append(closure)
+        return WorkflowView(isPresented: $isPresented,
+                            workflow: workflow,
+                            onFinish: onFinish,
+                            onAbandon: onAbandon,
+                            passedArgs: passedArgs)
+    }
+
+    /// Adds an action to perform when this `Workflow` has abandoned.
+    public func onAbandon(closure: @escaping () -> Void) -> Self {
+        var onAbandon = self.onAbandon
+        onAbandon.append(closure)
+        return WorkflowView(isPresented: $isPresented,
+                            workflow: workflow,
+                            onFinish: onFinish,
+                            onAbandon: onAbandon,
+                            passedArgs: passedArgs)
+    }
+}
+
+// MARK: thenProceeds
+@available(iOS 14.0, macOS 11, tvOS 14.0, watchOS 7.0, *)
+extension WorkflowView where Args == Never {
+    /**
+     Adds an item to the workflow; enforces the `FlowRepresentable.WorkflowOutput` of the previous item matches the args that will be passed forward.
+     - Parameter workflowItem: a `WorkflowItem` that holds onto the next `FlowRepresentable` in the workflow.
+     - Returns: a new `WorkflowView` with the additional `FlowRepresentable` item.
+     */
+    public func thenProceed<FR: FlowRepresentable & View>(with item: WorkflowItem<FR>) -> WorkflowView<FR.WorkflowOutput> where FR.WorkflowInput == Never {
+        var workflow = self.workflow
+        if workflow == nil {
```

Comment:
We should reverse this `if` statement so that appending is the primary path and initialization is the secondary.  Reasoning: because we expect this `thenProceed` to be appending more than it is initializing.

This of course applies to all the typed versions of this.

Reply:
Not sure I entirely agree. The way it reads now is based on what will happen chronologically, that makes more sense in my head. It starts as nil, then we must assign it, then it is not nil anymore.

Reply:
I was thinking in regards to the styleguide: [DO write if/else statements starting with the happy path.](https://github.com/wwt/SwiftCurrent/blob/main/.github/STYLEGUIDE.md#control-flow).  The happy path in this regard being appending, not initializing.  It would be something like:
```swift
if workflow != nil {
  workflow?.append(item.metadata)
} else {
  workflow = AnyWorkflow(Workflow<FR>(item.metadata))
}
```

But after looking at that and not being able to make it an `if let workflow = workflow`, I think I'm coming around that `if workflow == nil` is clearer.